app/modules/intelligence/provider/anthropic_caching_model.py

```
@@ -0,0 +1,695 @@
+                    for key, value in usage_dict.items()
+                    if isinstance(value, int)
+                }
+
+                if usage_details:
+                    _log_cache_metrics(usage_details, str(self._model_name))
+
+        # Process the streamed response normally (same as parent)
+        if isinstance(first_chunk, _utils.Unset):
+            raise UnexpectedModelBehavior(
+                "Streamed response ended without content or tool calls"
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)